src/utils/update/review.beneficial.owner.ts

```diff
@@ -0,0 +1,96 @@
+import { BeneficialOwnerGov } from "../../model/beneficial.owner.gov.model";
+import * as config from "../../config";
+import { ApplicationData } from "../../model";
+import { BeneficialOwnerIndividual } from "../../model/beneficial.owner.individual.model";
+
+const AllBoTypes = {
+  boiReview: "review_beneficial_owners_individual",
+  boGovReview: "review_beneficial_owners_government_or_public_authority",
+  boIndividual: "beneficial_owners_individual",
+  boGov: "beneficial_owners_government_or_public_authority"
+};
+
+const beneficialOwnerReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+      + config.UPDATE_REVIEW_BENEFICIAL_OWNER_INDIVIDUAL_PAGE
+      + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+const beneficialOwnerGovReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+        + config.UPDATE_REVIEW_BENEFICIAL_OWNER_GOV_PAGE
+        + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+const checkBoIndividualValidation = (boi: BeneficialOwnerIndividual): boolean => {
+  return boi?.usual_residential_address ? true : false;
+};
+
+const checkBoGovValidation = (boi: BeneficialOwnerGov): boolean => {
+  return boi?.principal_address ? true : false;
+};
+
+const checkForBackButtonBo = (appData: ApplicationData, boType: string, boRedirectUrl: string) => {
+  const boLength: number = appData[boType]?.length || 0;
+  const boIndex = boLength - 1;
+
+  if (((appData[boType] && boLength >= 1) && (boType === AllBoTypes.boIndividual) && !checkBoIndividualValidation(appData[boType][boIndex]))
+      || ((appData[boType] && boLength >= 1) && (boType === AllBoTypes.boGov) && !checkBoGovValidation(appData[boType][boIndex]))){
```

Comment:
Could this be re-written so `appData[boType] && boLength >= 1` is only needed once?

```if (
    ((appData[boType] && boLength >= 1) &&
      ((boType === AllBoTypes.boIndividual) && !checkBoIndividualValidation(appData[boType][boIndex]))
    || ((boType === AllBoTypes.boGov) && !checkBoGovValidation(appData[boType][boIndex]))
    )```

Reply:
Fixed